### PR TITLE
Jason add map

### PR DESCRIPTION
**Pull request by JasonMartin85**
Added map component files, the ReserveMap component, and basic map functionality.

### CODE REVIEW

File: package-lock.json
Comment (JacobTheEldest) [inline package-lock.json:1]: Please delete

File: package.json
Comment (JacobTheEldest) [inline package.json:1]: Please delete

File: client/src/Reserve/ReserveMap.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        </Controls>
+      </Map>
+      <div>
+        {geoArray.length > 0 ? geoArray.map((asset,index)=><div key={index}>{asset.properties.name}<input type="checkbox" onChange={()=>handleChange(asset.properties.name,asset.geometry.coordinates)}/></div>) : <></>}
+        
+      </div>
+    </div>
+  );
+};
+export default ReserveMap;
```
Comment (JacobTheEldest) [inline client/src/Reserve/ReserveMap.js:109]: newline

File: client/src/Map/Wrappers/xyz.js
Comment (JacobTheEldest) [inline client/src/Map/Wrappers/xyz.js:1]: newline

File: client/src/Map/Wrappers/wrapper.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import vector from './vector';
+import xyz from './xyz'
+import osm from './osm';
+
+export {
+  vector,
+  xyz,
+  osm
+}
```
Comment (JacobTheEldest) [inline client/src/Map/Wrappers/wrapper.js:9]: newline

File: client/src/Map/Wrappers/vector.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import {Vector as VectorSource } from 'ol/source';
+
+function vector({ features }) {
+  return new VectorSource({
+    features
+  });
+}
+
+export default vector;
```
Comment (JacobTheEldest) [inline client/src/Map/Wrappers/vector.js:9]: newline

File: client/src/Map/Wrappers/osm.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import * as olSource from 'ol/source';
+
+function osm() {
+  return new olSource.OSM();
+}
+
+export default osm;
```
Comment (JacobTheEldest) [inline client/src/Map/Wrappers/osm.js:7]: newline

File: client/src/Map/Map.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  }, [center])
+  return (
+    <MapContext.Provider value={{ map }}>
+      <div ref={mapRef} className="ol-map">
+        {children}
+      </div>
+    </MapContext.Provider>
+  )
+}
+export default Map;
```
Comment (JacobTheEldest) [inline client/src/Map/Map.js:40]: newline

File: client/src/Map/Map.css [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+  position: absolute;
+  background-color: rgba(255,255,255,0.4);
+  border-radius: 4px;
+  padding: 2px;
+}
+
+.ol-full-screen {
+  top: .5em;
+  right: .5em;
+}
```
Comment (JacobTheEldest) [inline client/src/Map/Map.css:19]: newline

File: client/src/Map/Layers/VectorLayer.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    return () => {
+      if (map) {
+        map.removeLayer(vectorLayer)
+      }
+    }
+  }, [map])
+  return null;
+}
+
+export default VectorLayer;
```
Comment (JacobTheEldest) [inline client/src/Map/Layers/VectorLayer.js:25]: newline

File: client/src/Map/Layers/TileLayer.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    return () => {
+      if (map) {
+        map.removeLayer(tileLayer)
+      }
+    }
+  }, [map])
+  return null;
+}
+
+export default TileLayer;
```
Comment (JacobTheEldest) [inline client/src/Map/Layers/TileLayer.js:25]: newline

File: client/src/Map/Layers/Layers.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+const Layers = ({children}) => {
+  
+  return( <div>{children}</div>)
+}
+
+export default Layers;
```
Comment (JacobTheEldest) [inline client/src/Map/Layers/Layers.js:8]: newline

File: client/src/Map/Layers/KMLVectorLayer.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+    return () => {
+      if (map) {
+        map.removeLayer(vectorLayer)
+      }
+    }
+  }, [map])
+  return null;
+}
+
+export default KMLVectorLayer;
```
Comment (JacobTheEldest) [inline client/src/Map/Layers/KMLVectorLayer.js:33]: newline

File: client/src/Map/Controls/FullScreenControl.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+    if (!map) return;
+    let fullScreenControl = new FullScreen({});
+    map.controls.push(fullScreenControl)
+
+    return () => map.controls.remove(fullScreenControl)
+  }, [map])
+  return null;
+}
+
+export default FullScreenControl
```
Comment (JacobTheEldest) [inline client/src/Map/Controls/FullScreenControl.js:17]: newline

File: client/src/Map/Controls/Controls.js [hunk 12]
```diff
@@ -0,0 +1,7 @@
+import React from 'react'
+
+const Controls = ({children}) => {
+  return <div>{children}</div>
+}
+
+export default Controls
```
Comment (JacobTheEldest) [inline client/src/Map/Controls/Controls.js:7]: newline

**Review by JacobTheEldest — CHANGES_REQUESTED** [review]

**Review by hackek — APPROVED** [review]
Yay

File: client/src/Map/Wrappers/xyz.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+import * as olSource from 'ol/source';
+
+function xyz({url, attributions, maxZoom}) {
+  return new olSource.XYZ({url, attributions, maxZoom})
+}
+
+export default xyz;
```
Comment (JacobTheEldest) [inline client/src/Map/Wrappers/xyz.js:7]: newline

**Review by JacobTheEldest — APPROVED** [review]

**Pull request merged**